### PR TITLE
add POST /api/articles/:article_id

### PR DESCRIPTION
**Pull request by liampjh34**
1. Add new endpoint - POST /api/articles/:article_id
2. Add new error handler to check votes vs. desired decrement
3. Add associated tests
4. Add new endpoint to endpoints.json

### CODE REVIEW

File: __tests__/articles.tests.js [hunk 1]
```diff
@@ -63,4 +63,83 @@ describe('/api/articles/:article_id', () => {
         .expect(404)
         expect(body.msg).toBe('No results found')
     });
+});
+
+describe('PATCH /api/articles/:article_id', () => {
+    it('decrementing votes should respond with 200 and the updated article', async () => {
```
Comment (LangaLee) [inline __tests__/articles.tests.js:69]: you are incrementing votes below so lets describe that 😄

File: __tests__/articles.tests.js [hunk 2]
```diff
@@ -63,4 +63,83 @@ describe('/api/articles/:article_id', () => {
...
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: 1,
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: 90,
+            article_img_url: expect.any(String)
+        })
+    });
+    it('should 403 when trying to decrement a vote count that is already 0', async () => {
```
Comment (LangaLee) [inline __tests__/articles.tests.js:105]: lets use a 400 here 😄

File: __tests__/articles.tests.js [hunk 3]
```diff
@@ -63,4 +63,83 @@ describe('/api/articles/:article_id', () => {
...
+        const input = {
+            inc_votes: 10
+        }
+        const { body } = await request(app)
+        .patch('/api/articles/3')
+        .send(input)
+        .expect(403)
+        expect(body.msg).toBe('Not allowed!')
+    });
+    it('should 403 if trying to decrement votes more than the vote count', async () => {
```
Comment (LangaLee) [inline __tests__/articles.tests.js:115]: lets use a 400 here also 😄

File: __tests__/error-handlers.tests.js [hunk 4]
```diff
@@ -76,4 +77,35 @@ describe('checkIsUser()', () => {
             })
         }
     })
+});
+
+describe('checkVotes()', () => {
+    it('should reject if desired decrement > votes count', async () => {
```
Comment (LangaLee) [inline __tests__/error-handlers.tests.js:83]: Is the test in the articles.test.js not testing this ?

**Review by LangaLee — COMMENTED** [review]
Nice work 😄 left some comments below

File: __tests__/error-handlers.tests.js (hunk 4 above)
Comment (liampjh34) [reply __tests__/error-handlers.tests.js:83]: It is! I thought we had to unit test our helper/error handling functions too - is it okay to skip that? 😊

**Pull request merged**